### PR TITLE
Remove sprite

### PR DESCRIPTION
**Pull request by piotrGTX**

### CODE REVIEW

File: core/src/main/kotlin/com/codattle/core/service/FileService.kt [hunk 1]
```diff
@@ -32,4 +32,9 @@ class FileService(private val storageService: StorageService, private val fileDa
             throw exception
         }
     }
+
+    fun deleteFile(fileId: Id<File>) {
+        storageService.delete(fileId.value)
```
Comment (piotrwalkusz1) [inline core/src/main/kotlin/com/codattle/core/service/FileService.kt]: Document from collection `file` should be removed too

File: core/src/main/resources/schema.graphqls [hunk 2]
```diff
@@ -112,6 +112,8 @@ type Mutation {
 
     addSpriteToGame(gameId: ID!, sprite: NewSprite!): Boolean!
 
+    removeSpriteFromGame(gameId: ID!, sprite: NewSprite!): Boolean!
```
Comment (piotrwalkusz1) [inline core/src/main/resources/schema.graphqls]: Type of argument shouldn't be `NewSprite`. What if a user prepares fake `NewSprite` with nonmatching `name` and `image`?  Only name of the sprite should be sent and the field `image` should be fetched from `Mongo`.

File: frontend/src/component/SpriteList.re [hunk 3]
```diff
@@ -71,23 +94,30 @@ let make =
         (
           uploadedSprites
           |> List.map(({name, fileId}) =>
-               <div key=name> <span> {ReasonReact.string(name)} </span> <img src={Environment.storageUrl ++ fileId} width height /> </div>
+               <div key=name>
+                 <span> {ReasonReact.string(name)} </span>
+                 <Button label="remove" onClick={() => removeUploaded(uploadedSprites, name)} />
```
Comment (piotrwalkusz1) [inline frontend/src/component/SpriteList.re]: Add new translation with key `common.remove`.

File: frontend/src/component/SpriteList.re [hunk 4]
```diff
@@ -71,23 +94,30 @@ let make =
...
              )
         )
         @ (
           notUploadedSprites
           |> List.map(({name, file}) =>
-               <div key=name> <span> {ReasonReact.string(name)} </span> <ImagePreview image=file width height /> </div>
+               <div key=name>
+                 <ImagePreview image=file width height />
+                 <div> {ReasonReact.string(name)} </div>
+                 <Button label="remove" onClick={() => removeNotUploaded(notUploadedSprites, name)} />
```
Comment (piotrwalkusz1) [inline frontend/src/component/SpriteList.re]: Add new translation with key `common.remove`.

File: core/src/main/kotlin/com/codattle/core/service/FileService.kt (hunk 1 above)
Comment (piotrGTX) [reply core/src/main/kotlin/com/codattle/core/service/FileService.kt]: Fixed

File: core/src/main/resources/schema.graphqls (hunk 2 above)
Comment (piotrGTX) [reply core/src/main/resources/schema.graphqls]: Fixed

File: frontend/src/component/SpriteList.re (hunk 3 above)
Comment (piotrGTX) [reply frontend/src/component/SpriteList.re]: Fixed

File: frontend/src/component/SpriteList.re (hunk 4 above)
Comment (piotrGTX) [reply frontend/src/component/SpriteList.re]: Fixed

**Pull request merged**